tests/functional/vault/test_shares.py

```diff
@@ -22,6 +27,59 @@ def test_deposit_with_wrong_amount(vault, token, gov):
         vault.deposit(balance, {"from": gov})
 
 
+def test_deposit_with_guest_list_not_invited(
+    vault, guestList, token, gov, rando, history
+):
+    # Make sure we're attempting to deposit something
+    token.transfer(rando, token.balanceOf(gov) // 2, {"from": gov})
+    balance = token.balanceOf(rando)
+    token.approve(vault, balance, {"from": rando})
+
+    # Note - don't need to call guestList.setGuests, since nobody's invited by
+    # default.
+    # gov is our bouncer
+    vault.setGuestList(guestList, {"from": gov})
+
+    # Ensure rando's not permitted to deposit
+    with brownie.reverts():
+        vault.deposit(balance, {"from": rando})
+
+    # Ensure authorized was called and that the deposit didn't revert for a
+    # different reason.
+    assert history[-1].subcalls[-1]["function"] == "authorized(address,uint256)"
+    assert history[-1].subcalls[-1]["return_value"][0] == False
+
+
+def test_deposit_with_guest_list_invited(vault, guestList, token, gov, rando, history):
```

Comment:
Can merge this test with the other test by first showing it reverts when they're not authorized, then they can deposit when they are. The setup is basically the same.

Reply:
Makes perfect sense, will do.